helix-term/src/args.rs

```diff
@@ -103,9 +103,9 @@ impl Args {
                     }
                 }
                 arg if arg.starts_with('+') => {
-                    match arg[1..].parse::<usize>() {
-                        Ok(n) => line_number = n.saturating_sub(1),
-                        _ => insert_file_with_position(arg),
+                    line_number = match arg[1..].parse::<usize>() {
+                        Ok(n) => n.saturating_sub(1),
+                        _ => usize::MAX,
```

Comment:
This is a bit too permissive: `+ABCD` would also cause the position to become the EOF. Instead, above this clause we can add a branch to the match that matches `"+"` explicitly. That would only apply to a lone `+` argument.